File: DeletewithoutHeadPointer.py
class Solution:
"""Time Complexity-O(1)
Space Complexity-O(1)"""
    #Function to delete a node without any reference to head pointer.
    def deleteNode(self,curr_node):
        removeNode=curr_node.next
        curr_node.data=curr_node.next.data
        curr_node.next=curr_node.next.next
        removeNode.next=None
        

File: Delete_without_head_pointer.cpp
/*
Time Complexity: O(1) 
Space Complexity: O(1)
Did this code successfully run on Leetcode :yes
Any problem you faced while coding this : no
*/

/*
struct Node {
  int data;
  struct Node *next;
  Node(int x) {
    data = x;
    next = NULL;
  }
}*head;
*/


class Solution
{
    public:
    //Function to delete a node without any reference to head pointer.
    void deleteNode(Node *del)
    {
       // Your code here
       del->data = del->next->data;
       del->next = del->next->next;
    }

};